src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -0,0 +1,51 @@
+package frc.robot.subsystems;
+
+import com.revrobotics.CANSparkLowLevel.MotorType;
+import com.revrobotics.CANSparkMax;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+
+public class ShooterSubsystem extends SubsystemBase {
+
+  private final CANSparkMax shooterMotor;
+  boolean readyToFire = true; // Take in readiness state from ShooterTrigger?
+
+  public ShooterSubsystem() {
+    shooterMotor =
+        new CANSparkMax(Constants.ShooterConstants.SHOOTER_MOTOR_ID, MotorType.kBrushless);
+    shooterMotor.restoreFactoryDefaults();
+    shooterMotor.setSmartCurrentLimit(40);
+  }
+
+  public Command shooterFire(final double distance) {
+    return startEnd(
+        () -> {
+          if (readyToFire) {
+            readyShooter(distance);
+          }
+        },
+        () -> {
+          setShooterSpeed(Constants.ShooterConstants.SHOOTER_MINIMUM_SPEED);
```

Comment:
Let's do this as a default command.  

https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj2/command/CommandScheduler.html#setDefaultCommand(edu.wpi.first.wpilibj2.command.Subsystem,edu.wpi.first.wpilibj2.command.Command)


So, this would be two steps:
One, a method to set the motor to the minimum speed.
Two, a call to setDEfaultCommand in RobotContainer for ShooterSubsystem.

Then we don't need to make sure to manage when we need to switch to the minimum speed.

Reply:
Resolved in commit b99820f.